GameModel.md

```diff
@@ -0,0 +1,78 @@
+
+The first taker of the contract will become Worker. However, the Verifier should be randomly chosen between a sufficiently large pool of volunteers who have made an Initial Deposit (not necessarily equal to v, can be smaller), so as to reduce the possibility of collusion between W, V or C, V.
+
+The full execution of the contract is as follows:
+
+
+Stage 1.
+---
+
+The Client and the Worker come to agreement about contract terms. The contract for Work is signed by (at least) C, W - this ends stage 1.
+
+Until the contract is signed, there are no further stages.
+
+
+Stage 2.
+---
+
+The Worker proceeds to calculate the Work. Then, a part of ready Work is chosen as a Sample for verification purposes (according to a known tech process not described here). The Sample is chosen in a way that the Worker does not know which part of Work will be checked and therefore cannot choose which part of Work to calculate selectively. An economically feasible size of such Sample is between 5% and 20% of Work. The Sample result is reported to the Client - this ends stage 2.
+
+If the Worker fails to send the Sample result by a certain time, then he is penalized (not necessarily with a full stake w), and a replacement Worker can take this contract. Alternatively, if that it easier, the contract could be dissolved and everyone except the Worker receives full refund of deposits, while the Worker is penalized (not necessarily with a full stake w). This penalty prevents the stalling attack. The money from penalty could be burnt or given to the Client (and/or Verifier, if he was already chosen) as a compensation for wasting time.
+
+If the Worker sees that data are invalid and cannot be used for calculations, then there are various options to deal with the problem. One easy (and harsh for the Client) option is to allow the Worker to proceed as normal with the contract, and the Verifier and Arbiters should consider any results reported by the Worker as correct, and Client must give the full amount of Payment for this Work. Another option is to introduce a stage 1.5 which allows the Worker to back off from the contract after seing the data (if this is allowed without penalties, it makes possible a stalling attack by the Worker, but if we add a small penalty, it makes possible a flooding attack by the Client, so neither solution is perfect). Another option is to have Worker and Verifier both agree on poor quality of data, in which case the Client can receive refund of Payment minus a small penalty (while if Verifier does not agree with Worker on poor quality of data, then either Arbitration, or the Worker can back off from the contract with some penalty).
```

Comment:
Nice review of the possible options. Am I right that these two options are the same in terms of Nash equilibriums?

> allow the Worker to proceed as normal with the contract, and the Verifier and Arbiters should consider any results reported by the Worker as correct, and Client must give the full amount of Payment for this Work.

> to have Worker and Verifier both agree on poor quality of data, in which case the Client can receive refund of Payment minus a small penalty (while if Verifier does not agree with Worker on the poor quality of data, then either Arbitration or the Worker can back off from the contract with some penalty).

Reply:
The first option appears much clearer and should lead to a provable Nash. The second option should, in my opinion, be considered only if the first option is impractical for "marketing" reason (it punishes Client's mistakes harshly). The second option has to be designed carefully (or maybe even modified, allowing the Client to go to Arbitration too), so as not to open the network to various minor attacks. It creates a tradeoff between burdening the Arbiters with the trivial task of data checking (if C, W, V do not all agree) and assigning minor penalties that cannot be protested in Arbitration.